language.md

```diff
@@ -625,3 +625,125 @@ seen on the status page, or:
+prepare this sample; if so, changing it might need to trigger a new action in
+Shesmu. If the input hasn't changed, even though Shesmu has been restarted, or
+the olive is changed in an unimportant way, the input to the action should be
+considered the same. A signature variable is a way to create a unique signature
+based on the data that was actually used by an olive that can be stored as part
+of the action.
+
+First, in every input format, some variables are considered _signable_. That
+means their values may be changed for the same ID. Fields which are immutable
+are not signable. The exact definition depends on the needs of the input
+format. Second, an olive references variables from the input and Shesmu tracks
+these variables.
+A variable is considered used if it is possible to be referenced, but not
+necessarily referenced. For instance:
+
+    False ? a : b
+
+references both `a` and `b` even though the value of `a` is never actually used.
+
+The _signable_ variables are used to create _signature_ variables. The
+supported signature variables are shown on the status page and apply to all
+import formats.
+
+Signature variables are treated like any other variable and they are lost
+during a `Group`, `Join`, `Smash` or `Let` operation if not preserved. Once the
+input is manipulated by a `Group`, `Join`, `Smash` or `Let` operation, it is
+not possible to track what input was used. So, references are only considered
+from the start of the olive until the first manipulation operation. If it is a
+`Run` olive, with no manipulation, the `With` arguments are included.
+
+The collection of included variables is considered over the whole scope. For
+instance:
+
+   Run x
+     Where "project" In signable_names # This is true even though project is
+                                       # referenced after this check
+     Where project ~ /N.*/
+     With { project = project }
+
+Since there are no `Group`, `Join`, `Smash` or `Let` clauses, the entire olive
+is in scope. The signable variable `project` is used twice: once in the `Where`
+clause and once in the arguments to the action. Therefore `project` is one of
+the referenced variables and appears in `signable_names` and its value will be
+in `sha1_signature`. Order does not matter: although `project` is referenced
+after `signature_names` is used, it is still present because it is used in that
+scope.
+
+This behaviour is necessary to ensure that `sha1_signature` returns the same
+value in all parts of the program.
+
+#### Example
+As an example, let's suppose we want to run a variant caller. It will take a
+list of genome alignments for different tumour/normal pairs in a patient and
+produce variant information. Whether the tissue is reference or normal is
+`tissue_type`. Now, the `tissue_type` is associated with each genome alignment
+(BAM) file, but if the tissue type is changed, the BAM file is not, so the
+action that produces the BAM will not be re-run. However, it affects the
```

Comment:
`will not be` -> `does not need to be` ? `However, it affects` -> `However, changing the tissue_type affects` ?